numba/np/arraymath.py

```diff
@@ -1810,6 +1810,52 @@ def np_roll_impl(a, shift):
         return np_roll_impl
 
 
+@overload(np.tile)
+def np_tile(a, reps):
+    def np_tile_integer_impl(a, reps):
+        if reps < 0:
+            raise ValueError("tile(): second argument must be non-negative")
+        arr = np.asarray(a)
+        sz = arr.size
+        out = np.empty(reps * sz, dtype=arr.dtype)
+        for i in range(reps):
+            start = i * sz
+            end = start + sz
+            out[start:end] = arr.flatten()
+        return out
+
+    def np_tile_impl(a, reps):
+        for r in reps:
+            if r < 0:
+                raise ValueError("tile(): second argument must be "
+                                 "non-negative")
+
+        arr = np.asarray(a)
+        sz = arr.size
+        # assumes arr is 1-dimensional
+        out_shape = reps[:-1] + (reps[-1] * sz,)
+        num_copies = reduce(lambda x, y: x * y, reps, 1)
+        out = np.empty(shape=num_copies * sz, dtype=arr.dtype)
+        for i in range(num_copies):
+            start = i * sz
+            end = start + sz
+            out[start:end] = arr.flatten()
```

Comment:
Again, better call `flatten()` once before the loop. Also, if you assume `arr` is 1D then why the use of `flatten()` at all?

Reply:
Good point. I was using `flatten()` with an eye towards supporting arrays of arbitrary dimension, which is held up because I haven't been able to figure out how to dynamically allocate a tuple of length `max(len(reps), arr.ndim)` for `out_shape`.